File: output/GPT3.5-Turbo/xCodeEval_4000/C++/Java/8289b8e19371f85195a43a39736a559b.java
import java.util.*;
import java.io.*;

class Main {
    static class FastScanner {
        BufferedReader br;
        StringTokenizer st;

        public FastScanner() {
            br = new BufferedReader(new InputStreamReader(System.in));
        }

        String next() {
            while (st == null || !st.hasMoreElements()) {
                try {
                    st = new StringTokenizer(br.readLine());
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
            return st.nextToken();
        }

        int nextInt() {
            return Integer.parseInt(next());
        }

        long nextLong() {
            return Long.parseLong(next());
        }

        double nextDouble() {
            return Double.parseDouble(next());
        }

        String nextLine() {
            String str = "";
            try {
                str = br.readLine();
            } catch (IOException e) {
                e.printStackTrace();
            }
            return str;
        }
    }

    static class FastPrinter {
        PrintWriter pw;

        public FastPrinter() {
            pw = new PrintWriter(System.out);
        }

        void print(String str) {
            pw.print(str);
        }

        void println(String str) {
            pw.println(str);
        }

        void println() {
            pw.println();
        }

        void flush() {
            pw.flush();
        }
    }

    static class ModInt {
        static int mod;
        int val;

        public ModInt(int val) {
            this.val = (val >= 0 ? val % mod : (mod - (-val) % mod) % mod);
        }

        ModInt add(ModInt p) {
            if ((val += p.val) >= mod) val -= mod;
            return this;
        }

        ModInt subtract(ModInt p) {
            if ((val += mod - p.val) >= mod) val -= mod;
            return this;
        }

        ModInt multiply(ModInt p) {
            val = (int) (1L * val * p.val % mod);
            return this;
        }

        ModInt divide(ModInt p) {
            return multiply(p.inverse());
        }

        ModInt negate() {
            return new ModInt(mod - val);
        }

        ModInt add(ModInt a, ModInt b) {
            return new ModInt(a.val).add(b);
        }

        ModInt subtract(ModInt a, ModInt b) {
            return new ModInt(a.val).subtract(b);
        }

        ModInt multiply(ModInt a, ModInt b) {
            return new ModInt(a.val).multiply(b);
        }

        ModInt divide(ModInt a, ModInt b) {
            return new ModInt(a.val).divide(b);
        }

        ModInt inverse() {
            int a = val, b = mod, u = 1, v = 0, t;
            while (b > 0) {
                t = a / b;
                int temp = a - t * b;
                a = b;
                b = temp;
                u -= t * v;
                int temp2 = u;
                u = v;
                v = temp2;
            }
            return new ModInt(u);
        }

        ModInt pow(long n) {
            ModInt ret = new ModInt(1);
            ModInt mul = new ModInt(val);
            while (n > 0) {
                if ((n & 1) == 1) {
                    ret.multiply(mul);
                }
                mul.multiply(mul);
                n >>= 1;
            }
            return ret;
        }

        static void setMod(int mod) {
            ModInt.mod = mod;
        }

        static int getMod() {
            return mod;
        }
    }

    static class DualSegTree {
        int n, log, size;
        ModInt[] laz;

        public DualSegTree(int n) {
            this.n = n;
            log = 1;
            while ((1 << log) < n) {
                ++log;
            }
            size = 1 << log;
            laz = new ModInt[size << 1];
            Arrays.fill(laz, new ModInt(0));
        }

        void reset() {
            Arrays.fill(laz, new ModInt(0));
        }

        void allApply(int k, ModInt a) {
            laz[k] = laz[k].add(a);
        }

        ModInt get(int p) {
            assert (0 <= p && p < n);
            p += size;
            for (int i = log; i >= 1; i--) {
                push(p >> i);
            }
            return laz[p];
        }

        ModInt[] getAll() {
            for (int i = 0; i < size; i++) {
                push(i);
            }
            return Arrays.copyOfRange(laz, size, size + n);
        }

        void apply(int l, int r, ModInt a) {
            assert (0 <= l && l <= r && r <= n);
            if (l == r) {
                return;
            }

            l += size;
            r += size;

            if (true) {
                for (int i = log; i >= 1; i--) {
                    if (((l >> i) << i) != l) {
                        push(l >> i);
                    }
                    if (((r >> i) << i) != r) {
                        push((r - 1) >> i);
                    }
                }
            }

            {
                int l2 = l, r2 = r;
                while (l < r) {
                    if ((l & 1) == 1) {
                        allApply(l++, a);
                    }
                    if ((r & 1) == 1) {
                        allApply(--r, a);
                    }
                    l >>= 1;
                    r >>= 1;
                }
                l = l2;
                r = r2;
            }
        }

        void debug() {
            System.out.println("dualsegtree getall:" + Arrays.toString(getAll()));
        }

        private void push(int k) {
            allApply(2 * k, laz[k]);
            allApply(2 * k + 1, laz[k]);
            laz[k] = new ModInt(0);
        }
    }

    static void solve() {
        int N = fs.nextInt();
        int mod = fs.nextInt();
        ModInt.setMod(mod);

        DualSegTree seg = new DualSegTree(N + 1);
        seg.apply(1, 2, new ModInt(1));

        for (int n = 1; n <= N; n++) {
            ModInt dp = seg.get(n);
            seg.apply(n + 1, N + 1, dp);
            int z = 2;
            while (true) {
                int l = n * z, r = (n + 1) * z;
                if (l > N) {
                    break;
                }
                ++z;
                r = Math.min(r, N + 1);
                seg.apply(l, r, dp);
            }
        }
        System.out.println(seg.get(N).val);
    }

    public static void main(String[] args) {
        FastScanner fs = new FastScanner();
        FastPrinter fp = new FastPrinter();

        int T = 1;

        for (int t = 0; t < T; t++) {
            solve();
        }

        fp.flush();
    }
}

File: output/GPT3.5-Turbo/xCodeEval_4000/C++/Go/8289b8e19371f85195a43a39736a559b.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

type modint struct {
	val int
}

func newModint(x int) modint {
	if x >= 0 {
		return modint{x % mod}
	}
	return modint{(mod - (-x) % mod) % mod}
}

func (m modint) add(p modint) modint {
	m.val += p.val
	if m.val >= mod {
		m.val -= mod
	}
	return m
}

func (m modint) sub(p modint) modint {
	m.val += mod - p.val
	if m.val >= mod {
		m.val -= mod
	}
	return m
}

func (m modint) mul(p modint) modint {
	m.val = (m.val * p.val) % mod
	return m
}

func (m modint) inv() modint {
	a := m.val
	b := mod
	u := 1
	v := 0
	for b > 0 {
		t := a / b
		a, b = b, a-t*b
		u, v = v, u-t*v
	}
	if u < 0 {
		u += mod
	}
	return modint{u}
}

func (m modint) pow(n int64) modint {
	ret := modint{1}
	mul := m
	for n > 0 {
		if n&1 == 1 {
			ret = ret.mul(mul)
		}
		mul = mul.mul(mul)
		n >>= 1
	}
	return ret
}

const mod = 1000000007

type dualSegTree struct {
	n    int
	log  int
	size int
	laz  []modint
}

func newDualSegTree(n int) *dualSegTree {
	log := 1
	for 1<<log < n {
		log++
	}
	size := 1 << log
	laz := make([]modint, size<<1)
	for i := range laz {
		laz[i] = modint{0}
	}
	return &dualSegTree{
		n:    n,
		log:  log,
		size: size,
		laz:  laz,
	}
}

func (t *dualSegTree) reset() {
	for i := range t.laz {
		t.laz[i] = modint{0}
	}
}

func (t *dualSegTree) allApply(k int, a modint) {
	t.laz[k] = t.laz[k].add(a)
}

func (t *dualSegTree) get(p int) modint {
	p += t.size
	for i := t.log; i >= 1; i-- {
		t.push(p >> i)
	}
	return t.laz[p]
}

func (t *dualSegTree) getAll() []modint {
	for i := 0; i < t.size; i++ {
		t.push(i)
	}
	return t.laz[t.size : t.size+t.n]
}

func (t *dualSegTree) apply(l, r int, a modint) {
	if l == r {
		return
	}

	l += t.size
	r += t.size

	if true {
		for i := t.log; i >= 1; i-- {
			if (l>>i)<<i != l {
				t.push(l >> i)
			}
			if ((r - 1) >> i) << i != r-1 {
				t.push((r - 1) >> i)
			}
		}
	}

	{
		l2 := l
		r2 := r
		for l < r {
			if l&1 == 1 {
				t.allApply(l, a)
				l++
			}
			if r&1 == 1 {
				r--
				t.allApply(r, a)
			}
			l >>= 1
			r >>= 1
		}
		l = l2
		r = r2
	}
}

func (t *dualSegTree) debug() {
	fmt.Println("dualsegtree getall:", t.getAll())
}

type groupAdd struct{}

func (g groupAdd) op(x, y modint) modint {
	return x.add(y)
}

func (g groupAdd) inverse(x modint) modint {
	return modint{-x.val}
}

func (g groupAdd) power(x modint, n int64) modint {
	return modint{int(n) * x.val}
}

func (g groupAdd) unit() modint {
	return modint{0}
}

func (g groupAdd) commute() bool {
	return true
}

func solve() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanWords)

	scanner.Scan()
	N, _ := strconv.Atoi(scanner.Text())

	seg := newDualSegTree(N + 1)
	seg.apply(1, 2, modint{1})

	for n := 1; n <= N; n++ {
		dp := seg.get(n)
		seg.apply(n+1, N+1, dp)
		z := 2
		for {
			l := n * z
			r := (n + 1) * z
			if l > N {
				break
			}
			z++
			if r > N+1 {
				r = N + 1
			}
			seg.apply(l, r, dp)
		}
	}
	fmt.Println(seg.get(N).val)
}

func main() {
	solve()
}
